presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushAggregationBelowOuterJoin.java

```diff
@@ -134,8 +153,56 @@ private static boolean groupsOnAllOuterTableColumns(AggregationNode node, JoinNo
 
         private static PlanNode getFirstNonIdentityNode(PlanNode node)
         {
-            PlanNodeSearcher searcher = new PlanNodeSearcher(node).where(planNode -> !(planNode instanceof ProjectNode) || !((ProjectNode) planNode).isIdentity());
-            return searcher.findFirst().orElseThrow(() -> new IllegalStateException("no nodes below identity project"));
+            return searchFrom(node)
+                    .where(planNode -> !(planNode instanceof ProjectNode) || !((ProjectNode) planNode).isIdentity())
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalStateException("no nodes below identity project"));
+        }
+
+        private static Optional<JoinNode> getFirstJoinNode(PlanNode node)
+        {
+            return searchFrom(node).skipOnlyWhen(planNode -> (planNode instanceof ProjectNode) && ((ProjectNode) planNode).isIdentity())
```

Comment:
make `.skinOnlyWhen` to be in separate line.

extract condition as method while it is used in two places.